index.bs

```diff
@@ -475,7 +475,8 @@ When this method is invoked, the user agent MUST execute the following algorithm
     1. If |extension| is not supported by this client platform, then [=continue=].
 
     1. Otherwise, let |result| be the result of running |extension|'s [=client processing=] algorithm on |argument|. If the
-        algorithm returned an error, [=continue=].
+        algorithm returned an error, return [=a promise rejected with=] a {{DOMException}} whose name is
+        "{{TypeError}}", and terminate this algorithm.
```

Comment:
Per your point 1, Vijay, if an extension is updated in an incompatible way, I'd argue that it should instead be a new extension.

Per your point 2, I'm good with clients throwing appropriate errors when they detect errors.

The point of this change is that it's, in general, a good architectural choice to enable errors to be caught and to fail early a the time they are caught.  Ignoring errors makes things hard to debug and allows incorrect code to appear to work, which is almost always a bad thing.  Fail fast.